recipes/chai3d/3.2.0/conanfile.py

```diff
@@ -0,0 +1,193 @@
+            arch_name = ("Win32", "x64")[self.settings.arch == "x86_64"]
+        elif self.settings.os == "Linux":
+            os_name = "lin"
+            arch_name = ("i686", "x86_64")[self.settings.arch == "x86_64"]
+        elif self.settings.os == "Macos":
+            os_name = "mac"
+            arch_name = ("i386", "x86_64")[self.settings.arch == "x86_64"]
+        return "{0}-{1}".format(os_name, arch_name)
+
+    def requirements(self):
+        # the following conan dependencies are missing:
+        # * core:
+        #  - giflib: available version (5.1.7) crashes during compilation
+        #  - lib3ds: not available in conan center
+        # * ode module:
+        #  - ode: not available in conan center
+        self.requires.add("eigen/3.3.7")
+        self.requires.add("glew/2.1.0@bincrafters/stable")
+        self.requires.add("libjpeg/9d")
+        self.requires.add("libpng/1.6.37")
+        self.requires.add("openal/1.19.1")
+        self.requires.add("pugixml/1.10@bincrafters/stable")
+        self.requires.add("theora/1.1.1@bincrafters/stable")
+        if self.options.with_bullet:
+            self.requires.add("bullet3/2.89")
+
+    def configure(self):
+        if self.settings.os not in ["Windows", "Linux", "Macos"]:
+            raise ConanInvalidConfiguration(
+                "Unsupported System. "
+                "This package only supports Windows/Linux/Macos")
+        elif (self.settings.os == "Windows"
+              and self.settings.compiler == "Visual Studio"
+              and tools.Version(self.settings.compiler.version) <= "8"):
+            raise ConanInvalidConfiguration(
+                "Unsupported Visual Studio version. "
+                "This package requires Visual Studio editions "
+                "2010 or later")
+        elif (self.settings.os == "Macos"
+              and tools.Version(self.settings.os.version) <= "10.8"):
+            raise ConanInvalidConfiguration(
+                "Unsupported Macos version. "
+                "This package requires Macos version "
+                "10.9 or later")
+        # chai3d builds with the std c++0x so we warn the user for a
+        # potential ABI incompatibility
+        if self.settings.compiler != "Visual Studio":
+            tools.check_min_cppstd(self, "11")
+        if (self.settings.compiler == "gcc"
+                and Version(self.settings.compiler.version) >= "5"
+                and self.settings.compiler.libcxx == "libstdc++"):
+            raise ConanInvalidConfiguration(
+                "Unsupported compiler configuration. "
+                "This package uses the c++0x standard and thus requires "
+                "compiler.libcxx=libstdc++11")
+
+    def config_options(self):
+        if self.options.shared or self.settings.os == "Windows":
+            del self.options.fPIC
+        # glew doesn't have an fPIC option, so its type must match with chai3d
+        self.options["glew"].shared = self.options.shared
```

Comment:
Maybe the comment was not accurate. The original `CMakeLists.txt` builds chai3d only as a static library and thus links against static glew.
I've tried to add the possibility of building both Chai3d library types (static and shared) by patching such CMakeLists.txt file, but linking shared chai3d against static glew results in an error (for mixing types):

> /usr/bin/ld: /home/conan/.conan/data/glew/2.1.0/bincrafters/stable/package/6ae331b72e7e265ca2a3d1d8246faf73aa030238/lib/libGLEW.a(glew.c.o): relocation R_X86_64_PC32 against symbol `__glewActiveTexture' can not be used when making a shared object; recompile with -fPIC

I've tried to fix this issue by matching lib types. Any suggestions to avoid forcing shared? Otherwise, I need to use chai3d's own glew dependency instead of conan's.